src/convertToObject.js

```diff
@@ -6,7 +7,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const rules = sourceString.split(';');
+
+  rules.forEach((rule) => {
+    const clearRule = rule.trim();
+
+    if (clearRule) {
+      const parts = clearRule.split(':');
+      const key = parts[0].trim();
+      const value = parts.length > 1 ? parts.slice(1).join(':').trim() : '';
+
+      if (key) {
+        result[key] = value;
+      }
+    }
+  });
+
+  return result;
```

Comment:
i think, that we can use `reduce` method for it. Let's try it
```suggestion
  return sourceString.split(';').reduce((style, declaration) => {
    
    ...........
    
    return style;
  }, {});
```